src/App/components/board/Board.tsx

```diff
@@ -39,70 +51,67 @@ export default function Board({
   playerTwo: Player;
   onRestart: () => void;
 }) {
-  const [board, setBoard] = useState<Symbol[]>(CELLS.map(() => ""));
+  const [board, setBoard] = useState<Character[]>(CELLS.map(() => ""));
   const [currentPlayer, setCurrentPlayer] = useState<Player>(playerOne);
   const [winner, setWinner] = useState<Player | null>(null);
-  const [gameOver, setGameOver] = useState<Boolean>(false);
+  const [complete, setComplete] = useState<Boolean>(false);
+  const isFull = board.every((cell) => cell !== "");
 
   const onReset = () => {
     setBoard(CELLS.map(() => ""));
     setCurrentPlayer(playerOne);
     setWinner(null);
+    setComplete(false);
   };
 
   const onClick = (index: number) => {
     setBoard((prevBoard) => {
       const newBoardState = [...prevBoard];
-      newBoardState[index] = currentPlayer.symbol;
-      currentPlayer.symbol === "X"
-        ? setCurrentPlayer(playerTwo)
-        : setCurrentPlayer(playerOne);
+      newBoardState[index] = currentPlayer.character;
+
+      if (currentPlayer.character === "X") {
+        setCurrentPlayer(playerTwo);
+      } else {
+        setCurrentPlayer(playerOne);
+      }
       return newBoardState;
     });
   };
 
-  const evaluate = useCallback(
-    (board: Symbol[]): void => {
-      let ws = evaluateRows(board);
-      if (!ws) ws = evaluateColumns(board);
-      if (!ws) ws = evaluateDiagonals(board);
-
-      if (ws && ws === playerOne.symbol) setWinner(playerOne);
-      if (ws && ws === playerTwo.symbol) setWinner(playerTwo);
-    },
-    [playerOne, playerTwo]
-  );
-
-  const isGameOver = (winner: Player | null, board: Symbol[]): void => {
-    setGameOver(!winner && board.indexOf("") === -1);
-  };
-
   useEffect(() => {
-    isGameOver(winner, board);
-  }, [winner, board]);
+    const wc = findWinner(board);
+    if (wc === playerOne.character) setWinner(playerOne);
+    if (wc === playerTwo.character) setWinner(playerTwo);
+  }, [board, playerOne, playerTwo]);
 
   useEffect(() => {
-    evaluate(board);
-  }, [board, evaluate]);
+    if (winner || isFull) {
```

Comment:
Nitpick:
This logic can move up to the useEffect above as we are already checking for the winner above:

```ts
useEffect(() => {
    const wc = findWinner(board);
    if (wc === playerOne.character) setWinner(playerOne);
    if (wc === playerTwo.character) setWinner(playerTwo);

   setComplete(wc !== null || isFull); // here I directly added the condition in the setState
  }, [board, playerOne, playerTwo, isFull]);
```